bee-ledger/src/workers/consensus/worker.rs

```diff
@@ -341,24 +345,46 @@ where
                                 // }
                             }
                             Err(reason) => {
-                                debug!("Snapshotting skipped: {:?}", reason);
+                                debug!("Snapshotting skipped: {reason}");
                             }
                         }
 
-                        match should_prune(&tangle, ledger_index, pruning_delay, &pruning_config) {
-                            Ok((start_index, target_index)) => {
-                                if let Err(e) =
-                                    prune::prune(&tangle, &storage, &bus, start_index, target_index, &pruning_config)
-                                        .await
-                                {
-                                    error!("Pruning failed: {:?}.", e);
-                                    // TODO: consider initiating an emergency (but graceful) shutdown instead of just
-                                    // panicking.
-                                    panic!("Aborting due to an unexpected pruning error.");
+                        match should_prune(&tangle, &storage, ledger_index, milestones_to_keep, &pruning_config) {
+                            Ok(pruning_task) => match pruning_task {
+                                PruningTask::ByIndexRange {
+                                    start_index,
+                                    target_index,
+                                } => {
+                                    if let Err(e) = prune::prune_by_range(
+                                        &tangle,
+                                        &storage,
+                                        &bus,
+                                        start_index,
+                                        target_index,
+                                        &pruning_config,
+                                    )
+                                    .await
+                                    {
+                                        error!("Pruning milestone range failed: {e}.");
```

Comment:
If I remember correclty we agreed on lowercase errors and without the "." at the end

Reply:
Panic messages, yes. Log entries however are still proper sentences.